### ISSUE TITLE
.coafile not showing the correct error

### ISSUE THREAD

**Issue by ishanSrt** [coala/coala#5044]
An earlier typo in .coafile which is being fixed by https://github.com/coala/coala/issues/5043 doesn't show the correct error message. Instead it shows an error based on the previous line
eg. if your .coafile has this 
```
use_spaces = true
default_action: **: ApplyPatchAction
```
it will show
```
 Unable to convert parameter 'use_spaces' into type <class 'bool'>.
```
 if your .coafile is 
```
files = *.txt
default_action: **: ApplyPatchAction
```
It will show **Unable to find that file/ No matching file**

or if your .coafile is 
```
bears = SpaceConsistencyBear
default_action: **: ApplyPatchAction
```
it will show 
```
No bears matching 'SpaceConsistencyBear
default_action: **: ApplyPatchAction' were found. Make sure you have coala-bears installed or you have typed the name correctly.
```
Instead it should point out the error

**Comment by ishanSrt** [comment]
I would like to self assign myself to this issue 😌

**Comment by ishanSrt** [comment]
@Makman2 mark blocked by https://github.com/coala/coala/issues/5043

**Comment by Makman2** [comment]
Oh I'm just thinking: The parser allows to continue values on next lines without specifying escape characters. This is probably why it points out that the error lies in the previous lines, because this is where the setting-assignment effectively begins, and it thinks those new malformed setting-syntaxes belong to the value.

**Comment by ishanSrt** [comment]
Yes exactly! That's what I could make out from https://github.com/coala/coala/issues/5044#issue-286363165
Btw how does the parser know that a line is complete?

**Comment by ishanSrt** [comment]
@Makman2 even in case of valid lines

**Comment by Makman2** [comment]
I guess it tries to match a pattern that indicates a new value (roughly `.*=.*`). If it does not match, it assumes the next line to be a value for the last setting parsed.

**Comment by Makman2** [comment]
I'm really not sure what to do now. An escape character would be probably nicest for multi-line values, but breaks coafiles quite badly...

Oh but just thinking: What about noting in case of errors, where the relevant lines are? For our case, it would display `lines 1-2`. Then people probably grasp the cause earlier.

**Comment by ishanSrt** [comment]
@Makman2 didn't get what you meant by this:
>What about noting in case of errors, where the relevant lines are? For our case, it would display lines 1-2. Then people probably grasp the cause earlier.

**Comment by ishanSrt** [comment]
I get if the line doesn't match `.*=.*`, it assumes the current line to be a part of the the previous line's setting. Can't see how to point out the correct error message in cases like these. My PR works for only a particular case but there can be so many cases when the current line won't be a valid setting for the previous line.

**Comment by Makman2** [comment]
For example
```
.coafile lines 1-3:
Can't convert setting ... to <type bool>.
```
So we additionally display in error message this `.coafile lines 1-3` message that allows the user to find the relevant location :)

**Comment by ishanSrt** [comment]
@Makman2 OK so I will implement exactly that.

**Comment by ishanSrt** [comment]
I would like to be reassigned to this

**Comment by akshatkarani** [comment]
I would like to work on this issue

**Comment by ishanSrt** [comment]
@akshatkarani as pointed out by @jayvdb on my PR and as suggested by @Makman2 it would be best to just print out the line numbers as one can always create a scenario where your solution fails to detect something similar.

**Comment by akshatkarani** [comment]
> @akshatkarani as pointed out by @jayvdb on my PR and as suggested by @Makman2 it would be best to just print out the line numbers as one can always create a scenario where your solution fails to detect something similar.

Okay

**Comment by ishanSrt** [comment]
This issue seems a tad rather difficult to implement in the source code. Should be labeled `difficulty/medium`

**Comment by akshatkarani** [comment]
In the initial PR we found starting line number of a setting. Now we need to find ending line number of a setting.
You said that `SourcePosition` needs to given as `SourceRange`? Can you explain a little on what to do?

**Comment by jayvdb** [comment]
improve Setting so that `Setting(source=SourceRange(...))` works, improve the ConfParser to collect start and end lines of each setting, and instead of using `SourcePosition`, use `SourceRange`.

**Comment by jayvdb** [comment]
So now the parser knows the start and end of most lines, work could begin on this task.

However iirc, there are still a few problems which were found in the parsing, especially when nested sections were used.  You could work on those if you wound prefer.

**Comment by akshatkarani** [comment]
There are some issues related to parsing, but if they are not blocking this issue, I would like to close this issue first. Please suggest what need's to be done ahead.
